models/sushiswap/optimism/sushi_optimism_trades.sql

```diff
@@ -0,0 +1,104 @@
+{{ config(
+    alias = 'trades'
+    ,partition_by = ['block_date']
+    ,materialized = 'incremental'
+    ,file_format = 'delta'
+    ,incremental_strategy = 'merge'
+    ,unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index', 'trace_address']
+    ,post_hook='{{ expose_spells(\'["optimism"]\',
+                                      "project",
+                                      "sushiswap",
+                                    \'["hosuke","codingsh"]\') }}'
+    )
+}}
+
+{% set project_start_date = '2021-03-03' %}
+
+WITH sushiswap_dex AS (
+    SELECT  t.evt_block_time                                             AS block_time,
+            `to`                                                         AS taker,
+            sender                                                       AS maker,
+            CASE WHEN amount0Out = 0 THEN amount1Out ELSE amount0Out END AS token_bought_amount_raw,
+            CASE WHEN amount0In = 0 THEN amount1In ELSE amount0In END    AS token_sold_amount_raw,
+            cast(NULL as double)                                         AS amount_usd,
+            CASE WHEN amount0Out = 0 THEN token1 ELSE token0 END         AS token_bought_address,
+            CASE WHEN amount0In = 0 THEN token1 ELSE token0 END          AS token_sold_address,
+            t.contract_address                                           AS project_contract_address,
+            t.evt_tx_hash                                                AS tx_hash,
+            ''                                                           AS trace_address,
+            t.evt_index
+    FROM {{ source('sushi_optimism', 'ConstantProductPool_evt_Swap') }} t
+    INNER JOIN {{ source('sushi_optimism', 'UniswapV2Factory_evt_PairCreated') }} f
+        ON f.pair = t.contract_address
+    {% if is_incremental() %}
+    WHERE t.evt_block_time >= date_trunc("day", now() - interval '1 week')
+    {% else %}
+    WHERE t.evt_block_time >= '{{ project_start_date }}'
+    {% endif %}
```

Comment:
`sushi_optimism_trades` is somehow different from sushiswap on other chains. The approach using `sushi_optimism.UniswapV2Factory_evt_PairCreated` may not work in this situation. You can try another approach, using `sushi_optimism.StablePool_evt_Swap`, and find out the pools' information from other factory contract.

Reply:
I'll create a new pull request using this tip, thanks.